Automattic/sensei
# Fix passmark issues

Fixes #3429, Fixes #3421, Fixes #3428, Fixes #3427

### Changes proposed in this Pull Request

* Allow decimal passmarks.
* Fix validation passmark/passmark percentage.
* Validate if passmark is between 0 and 100.

### Testing instructions

#3429
* Import the following CSV:
```
ID,Lesson,Slug,Description,Excerpt,Status,Course,Module,Prerequisite,Preview,Tags,Image,Length,Complexity,Video,Pass Required,Passmark,Number of Questions,Random Question Order,Auto-Grade,Quiz Reset,Allow Comments,Questions
,Decimal,,,,,,,,,,,,,,1,1.5,,,,,,
```
* Make sure the decimal was imported correctly.

#3421
* Import the following CSV:
```
ID,Lesson,Slug,Description,Excerpt,Status,Course,Module,Prerequisite,Preview,Tags,Image,Length,Complexity,Video,Pass Required,Passmark,Number of Questions,Random Question Order,Auto-Grade,Quiz Reset,Allow Comments,Questions
,Lesson – true,,,,,,,,true,,,,,,true,,,true,true,true,true,
```
* Make sure the lesson was fully imported with _Pass required to complete lesson_ selected and _Quiz passmark percentage_ set to 0.

#3428
* Import the following CSV:
```
ID,Lesson,Slug,Description,Excerpt,Status,Course,Module,Prerequisite,Preview,Tags,Image,Length,Complexity,Video,Pass Required,Passmark,Number of Questions,Random Question Order,Auto-Grade,Quiz Reset,Allow Comments,Questions
,Negative,,,,,,,,,,,,,,1,-1,,,,,,
,Out of Range,,,,,,,,,,,,,,1,101,,,,,,
```
* Make sure the lessons were partial imported with the error: `Passmark must be between 0 and 100.`.

#3427
* Import the following CSV:
```
ID,Lesson,Slug,Description,Excerpt,Status,Course,Module,Prerequisite,Preview,Tags,Image,Length,Complexity,Video,Pass Required,Passmark,Number of Questions,Random Question Order,Auto-Grade,Quiz Reset,Allow Comments,Questions
,High,,,,,,,,,,,,,,,100,,,,,,
,Negative,,,,,,,,,,,,,,,-1,,,,,,
,Out of Range,,,,,,,,,,,,,,,101,,,,,,
,Decimal,,,,,,,,,,,,,,,1.5,,,,,,
```
* Make sure all lessons were partially imported with _Pass Required_ unchecked and _Passmark_ not set. And you should see the following warning for all lessons - "Both Passmark and Pass Required should be supplied."
